src/holon/models/rule_actions/rule_action_add_set.py

```diff
@@ -176,6 +180,101 @@ def add_or_set_items(
+        base_parent_type = filtered_repository.base_model_type  # TODO make sure repository has this
+        try:
+            parent_fk_field_name = next(
+                parent_fk_fieldname
+                for parent_type, parent_fk_fieldname in RuleActionUtils.get_parent_classes_and_field_names(
+                    self.model_to_add.__class__
+                )
+                if base_parent_type == parent_type
+            )
+        except:
+            raise ValueError(
+                f"Type {base_parent_type} in the filter does not match found parent type {RuleActionUtils.get_parent_classes_and_field_names(self.model_to_add.__class__)} for model type {self.model_to_add.__class__.__name__}"
+            )
+
+        objects_added = 0
+
+        # NOT NECESSARY ANYMORE? PLS REVIEW - TAVM
+        # # get cloned contractscope
+        # if self.contract_to_add:
+        #     scenario = filtered_queryset[0].payload
+        #     old_contract_scope = self.contract_to_add.contractScope
+
+        #     cloned_contract_scope = Actor.objects.filter(
+        #         payload=scenario, original_id=old_contract_scope.id
+        #     ).first()
+
+        # only take first n objects
+        for filtererd_object in filtered_repository.all():
+
+            if reset_models_before_add:
+                # check which objects of the same type as model_to_add are already
+                # connected to our current filtered_object, and remove these
+
+                model_to_add_repository = scenario_aggregate.get_repository_for_model_type(
+                    self.model_to_add_base_class.__name__,
+                    model_subtype_name=self.model_to_add.__class__,
+                )
+
+                objects_to_delete = model_to_add_repository.filter_attribute_value(
+                    f"{parent_fk_field_name}_id",
+                    filtererd_object.id,
+                    AttributeFilterComparator.EQUAL,
+                )
+
+                for object in objects_to_delete:
+                    scenario_aggregate.remove_object(object)
+
+            # Only check < n for set count
+            if not reset_models_before_add or objects_added < n:
+                # add model_to_add to filtered object
+
+                if self.contract_to_add:
+                    scenario_aggregate.add_object(
+                        self.model_to_add,
+                        self.model_to_add_base_class,
+                        {
+                            parent_fk_field_name: filtererd_object,
+                            # "contractScope": cloned_contract_scope,
+                            "is_rule_action_template": False,
+                        },
+                    )
+
+                else:
+                    scenario_aggregate.add_object(
+                        self.model_to_add,
+                        self.model_to_add_base_class,
+                        {
+                            parent_fk_field_name: filtererd_object,
```

Comment:
For serialization we look at the xxx_id fields set with an id. Maybe we should update both here, so the whole object is up-to-date.
What do you think?